qcodes/dataset/descriptions/detect_shapes.py

```diff
@@ -0,0 +1,104 @@
+from typing import Union, Sequence, Dict, Any, Tuple, Optional, Sized
+from numbers import Integral
+from collections import abc
+
+import numpy as np
+
+from qcodes.instrument.parameter import (_BaseParameter,
+                                         ArrayParameter,
+                                         MultiParameter,
+                                         ParameterWithSetpoints)
+from qcodes.utils.validators import Arrays
+
+
+def get_shape_of_measurement(
+        parameters: Sequence[_BaseParameter],
+        steps: Union[Sequence[int], Sequence[Sized]] = ()
+) -> Dict[str, Tuple[int, ...]]:
+    """
+    Construct the shape of a measurement of a dependent parameter from the
+    parameter and the axes it is to be sweept over.
+
+    Args:
+        parameters: The dependent parameters to construct the shape for
+        *steps: Zero or more dimensions that the parameter is to be sweept over.
+           These can be given either as the length of the dimension or as
+           a sequence of numbers to sweep over.
+
+    Returns:
+        A dictionary from the parameter name to a tuple of integers describing
```

Comment:
a general question - are we following numpy's convention for the order of these integers that describe the shape? as in from outer dimension into inner (or reverse)?

Reply:
The data should be c continious since that is the numpy default that meas that the fasted changing axis needs to be the last one. I think this should be the other way around but need to double check

Reply:
So in summary the steps are now documented to come in the order from outermost to inner most and the array shapes are appended at the end. Remaining work to check that multidim arrays (Array Parameters, MultiParameters and Parameters with setpoint) are also correct